core/src/main/java/nl/nn/adapterframework/pipes/FixedResultPipe.java

```diff
@@ -167,7 +167,7 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 			}
 		}
 
-		message.closeOnCloseOf(session); // avoid connection leaking when the message itself is not consumed.
+		message.closeOnCloseOf(session, this); // avoid connection leaking when the message itself is not consumed.
```

Comment:
Pleister! This should be registered as closeOnCloseOf on the cause, not the effect!